pulsar-function-go/pf/instance.go

```diff
@@ -0,0 +1,289 @@
+	gi.setupClient()
+	if int32(gi.context.InstanceConf.FuncDetails.Source.SubscriptionType) == pb.SubscriptionType_value["FAILOVER"] {
+		consumerType = pulsar.Failover
+	}
+
+	funcDetails := gi.context.InstanceConf.FuncDetails
+	subscriptionName := funcDetails.Tenant + "/" + funcDetails.Namespace + "/" + funcDetails.Name
+
+	properties := getProperties(getDefaultSubscriptionName(
+		funcDetails.Tenant,
+		funcDetails.Namespace,
+		funcDetails.Name), gi.context.InstanceConf.InstanceID)
+
+	channel := make(chan pulsar.ConsumerMessage)
+
+	for topic, consumerConf := range funcDetails.Source.InputSpecs {
+		log.Debugf("Setting up consumer for topic: %s with subscription name: %s", topic, subscriptionName)
+		if consumerConf.ReceiverQueueSize != nil {
+			if consumerConf.IsRegexPattern {
+				consumer, err := gi.client.Subscribe(pulsar.ConsumerOptions{
+					TopicsPattern:     topic,
+					ReceiverQueueSize: int(consumerConf.ReceiverQueueSize.Value),
+					SubscriptionName:  subscriptionName,
+					Properties:        properties,
+					Type:              consumerType,
+					MessageChannel:    channel,
+				})
+				if err != nil {
+					log.Errorf("create consumer error:%s", err.Error())
+				}
+				gi.consumers = append(gi.consumers, consumer)
+			} else {
+				consumer, err := gi.client.Subscribe(pulsar.ConsumerOptions{
+					Topic:             topic,
+					SubscriptionName:  subscriptionName,
+					Properties:        properties,
+					Type:              consumerType,
+					ReceiverQueueSize: int(consumerConf.ReceiverQueueSize.Value),
+					MessageChannel:    channel,
+				})
+				if err != nil {
+					log.Errorf("create consumer error:%s", err.Error())
+				}
+
+				gi.consumers = append(gi.consumers, consumer)
+			}
+		} else {
+			if consumerConf.IsRegexPattern {
+				consumer, err := gi.client.Subscribe(pulsar.ConsumerOptions{
+					TopicsPattern:    topic,
+					SubscriptionName: subscriptionName,
+					Properties:       properties,
+					Type:             consumerType,
+					MessageChannel:   channel,
+				})
+				if err != nil {
+					log.Errorf("create consumer error:%s", err.Error())
+				}
+				gi.consumers = append(gi.consumers, consumer)
+			} else {
+				consumer, err := gi.client.Subscribe(pulsar.ConsumerOptions{
+					Topic:            topic,
+					SubscriptionName: subscriptionName,
+					Properties:       properties,
+					Type:             consumerType,
+					MessageChannel:   channel,
+				})
+				if err != nil {
+					log.Errorf("create consumer error:%s", err.Error())
+				}
+				gi.consumers = append(gi.consumers, consumer)
+			}
+		}
+		gi.context.InputTopics = append(gi.context.InputTopics, topic)
+
+	CLOSE:
+		for {
+			select {
+			case cm := <-channel:
+				gi.msg = append(gi.msg, cm.Message)
+			case <-time.After(time.Millisecond * 500):
+				close(channel)
+				break CLOSE
+			}
+		}
+	}
+
+	gi.actualExecution(function)
+	gi.close()
+}
+
+func (gi *GoInstance) actualExecution(function Function) {
```

Comment:
remove this function since the main logic is already in startFunction

Reply:
done